### ISSUE TITLE
@sveltejs/adapter-static 1.0.0-next.4 fails to build the hn.svelte.dev example

### ISSUE THREAD

**Issue by phaleth** [sveltejs/kit#864]
When deploying the hn.svelte.dev example locally using adapter-static an error occurs as shown in the output of the following procedure.

```shell
user@debian:~/kit$ cd examples/hn.svelte.dev
user@debian:~/kit/examples/hn.svelte.dev$ cat package.json
{
        "name": "hn.svelte.dev",
        "version": "1.0.0",
        "private": true,
        "type": "module",
        "scripts": {
                "dev": "svelte-kit dev",
                "build": "svelte-kit build",
                "start": "svelte-kit start --host localhost"
        },
        "devDependencies": {
                "@sveltejs/adapter-static": "next",
                "@sveltejs/kit": "next",
                "node-fetch": "^2.6.1",
                "rollup": "^2.44.0",
                "svelte": "^3.37.0",
                "vite": "^2.1.5"
        }
}
user@debian:~/kit/examples/hn.svelte.dev$ cat svelte.config.cjs
const ad = require('@sveltejs/adapter-static');

module.exports = {
        kit: {
                adapter: ad(),
                target: '#svelte'
        }
};
user@debian:~/kit/examples/hn.svelte.dev$ npx pnpm i
npx: installed 1 in 1.267s
Scope: all 12 workspace projects
../../packages/kit                       |  WARN  node-fetch: fetch-blob@2.1.1 requires a peer of domexception@^2.0.1 but none was installed.
../../packages/kit                       |  WARN  @sveltejs/vite-plugin-svelte@1.0.0-next.5 requires a peer of vite@^2.1.2 but version 2.1.0 was installed.
../..                                    |  WARN  @sveltejs/eslint-config@5.6.0 requires a peer of eslint-plugin-node@>= 11 but none was installed.
../..                                    |   +4 +
../..                                    | Progress: resolved 151, reused 150, downloaded 1, added 4, done
 WARN  Failed to find "/fsevents/2.3.2" in lockfile during hoisting. Next aliases will not be hoisted: fsevents

devDependencies:
+ @sveltejs/adapter-static 1.0.0-next.4 <- ../../packages/adapter-static
+ @sveltejs/kit 1.0.0-next.71
+ node-fetch 2.6.1
+ rollup 2.44.0
+ svelte 3.37.0
+ vite 2.1.5
user@debian:~/kit/examples/hn.svelte.dev$ npx pnpm -r build
npx: installed 1 in 1.268s
Scope: all 12 workspace projects
../.. build$ pnpm build --filter="@sveltejs/*" --filter="create-svelte"
│ Scope: 8 of 12 workspace projects
│ packages/kit build$ rm -rf assets/runtime && rollup -c
│ packages/kit build: src/runtime/client/start.js, src/runtime/client/singletons.js, src/runtime/app/navigation.js, src/runtime/app/stores.js, src/runtime/app/paths.js, sr
│ packages/kit build: created assets/runtime in 190ms
│ packages/kit build: src/cli.js, src/runtime/server/index.js, src/core/filesystem/index.js, src/core/http/index.js → dist...
└─ Running...
. build$ svelte-kit build
│ vite v2.1.5 building for production...
│ transforming...
│ ✓ 35 modules transformed.
│ rendering chunks...
│ .svelte/output/client/_app/manifest.json                                    1.81kb
[1 lines collapsed]
│ .svelte/output/client/_app/pages/about.svelte-772d4312.js                   2.96kb / brotli: 1.05kb
│ .svelte/output/client/_app/pages/user/[name].svelte-4e97fd38.js             2.91kb / brotli: 1.19kb
│ .svelte/output/client/_app/pages/item/[id].svelte-09ba3ebc.js               6.71kb / brotli: 2.21kb
│ .svelte/output/client/_app/pages/[list]/[page].svelte-f95225fd.js           6.05kb / brotli: 2.16kb
└─ Running...
[12 lines collapsed]uild$ rm -rf assets/runtime && rollup -c
│ .svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
│ .svelte/output/client/_app/chunks/vendor-b3a70cfd.js                        5.95kb / brotli: 2.28kb
│ .svelte/output/client/_app/start-8804a1d2.js                                23.38kb / brotli: 7.30kb...
│ vite v2.1.5 building SSR bundle for production...
│ transforming...
│ ✓ 34 modules transformed.
[1 lines collapsed]
│ packages/kit build: (!) Unused external imports
└─ Running...
. build$ svelte-kit buildortstatic
[12 lines collapsed]
│ .svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
│ .svelte/output/client/_app/chunks/vendor-b3a70cfd.js                        5.95kb / brotli: 2.28kb
│ .svelte/output/client/_app/start-8804a1d2.js                                23.38kb / brotli: 7.30kb
│ vite v2.1.5 building SSR bundle for production...
│ transforming...
│ ✓ 34 modules transformed.js in 10ms
│ rendering chunks...
│ .svelte/output/server/app.js   110.70kb
│ Run npm start to try your app locally.
│ > Using @sveltejs/adapter-static
└─ Running...
../../packages/kit build$ rm -rf assets/runtime && rollup -c
[1 lines collapsed]
│ packages/kit build: default imported from external module 'url' but never used
└─ Running...
. build$ svelte-kit buildorts
[12 lines collapsed]
│ .svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
│ .svelte/output/client/_app/chunks/vendor-b3a70cfd.js                        5.95kb / brotli: 2.28kb
│ .svelte/output/client/_app/start-8804a1d2.js                                23.38kb / brotli: 7.30kb
│ vite v2.1.5 building SSR bundle for production...
│ transforming...
│ ✓ 34 modules transformed.js in 10ms
│ rendering chunks...
│ .svelte/output/server/app.js   110.70kb
│ Run npm start to try your app locally.
│ > Using @sveltejs/adapter-static
└─ Running...
../../packages/kit build$ rm -rf assets/runtime && rollup -c
[1 lines collapsed]
│ packages/kit build: default imported from external module 'querystring' but never used
└─ Running...
. build$ svelte-kit buildorts
[12 lines collapsed]
│ .svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
│ .svelte/output/client/_app/chunks/vendor-b3a70cfd.js                        5.95kb / brotli: 2.28kb
│ .svelte/output/client/_app/start-8804a1d2.js                                23.38kb / brotli: 7.30kb
│ vite v2.1.5 building SSR bundle for production...
│ transforming...
│ ✓ 34 modules transformed.js in 10ms
│ rendering chunks...
│ .svelte/output/server/app.js   110.70kb
│ Run npm start to try your app locally.
│ > Using @sveltejs/adapter-static
└─ Running...
../../packages/kit build$ rm -rf assets/runtime && rollup -c
[1 lines collapsed]
│ packages/kit build: (!) Circular dependency
└─ Running...
. build$ svelte-kit buildorts
[12 lines collapsed]
│ .svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
│ .svelte/output/client/_app/chunks/vendor-b3a70cfd.js                        5.95kb / brotli: 2.28kb
│ .svelte/output/client/_app/start-8804a1d2.js                                23.38kb / brotli: 7.30kb
│ vite v2.1.5 building SSR bundle for production...
│ transforming...
│ ✓ 34 modules transformed.js in 10ms
│ rendering chunks...
│ .svelte/output/server/app.js   110.70kb
│ Run npm start to try your app locally.
│ > Using @sveltejs/adapter-static
└─ Running...
../../packages/kit build$ rm -rf assets/runtime && rollup -c
[1 lines collapsed]
│ packages/kit build: src/runtime/server/index.js -> src/runtime/server/page/index.js -> src/runtime/server/index.js
└─ Running...
. build$ svelte-kit buildorts
[12 lines collapsed]
│ .svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
│ .svelte/output/client/_app/chunks/vendor-b3a70cfd.js                        5.95kb / brotli: 2.28kb
│ .svelte/output/client/_app/start-8804a1d2.js                                23.38kb / brotli: 7.30kb
│ vite v2.1.5 building SSR bundle for production...
│ transforming...
│ ✓ 34 modules transformed.js in 10ms
│ rendering chunks...
│ .svelte/output/server/app.js   110.70kb
│ Run npm start to try your app locally.
│ > Using @sveltejs/adapter-static
└─ Running...
../../packages/kit build$ rm -rf assets/runtime && rollup -c
[1 lines collapsed]
[1 lines collapsed]
│ src/cli.js, src/runtime/server/index.js, src/core/filesystem/index.js, src/core/http/index.js → dist...
│ (!) Unused external imports
│ default imported from external module 'url' but never used
│ default imported from external module 'querystring' but never used
│ (!) Circular dependency
│ src/runtime/server/index.js -> src/runtime/server/page/index.js -> src/runtime/server/index.js
│ created dist in 4.7s
│ src/core/filesystem/index.js → dist/filesystem.cjs...
│ created dist/filesystem.cjs in 10ms
│ packages/kit build: created dist in 4.6s
└─ Running...
. build$ svelte-kit build
[12 lines collapsed]
│ .svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
│ .svelte/output/client/_app/chunks/vendor-b3a70cfd.js                        5.95kb / brotli: 2.28kb
│ .svelte/output/client/_app/start-8804a1d2.js                                23.38kb / brotli: 7.30kb
│ vite v2.1.5 building SSR bundle for production...
│ transforming...
│ ✓ 34 modules transformed.
│ rendering chunks...
│ .svelte/output/server/app.js   110.70kb
│ Run npm start to try your app locally.
│ > Using @sveltejs/adapter-static
└─ Running...
../../packages/kit build$ rm -rf assets/runtime && rollup -c
[1 lines collapsed]
│ vite v2.1.5 building for production...
│ transforming...
│ ✓ 35 modules transformed.
│ rendering chunks...
│ .svelte/output/client/_app/manifest.json                                    1.81kb
│ .svelte/output/client/_app/pages/index.svelte-887e5433.js                   0.21kb / brotli: 0.14kb
│ .svelte/output/client/_app/pages/about.svelte-772d4312.js                   2.96kb / brotli: 1.05kb
│ .svelte/output/client/_app/pages/user/[name].svelte-4e97fd38.js             2.91kb / brotli: 1.19kb
│ .svelte/output/client/_app/pages/item/[id].svelte-09ba3ebc.js               6.71kb / brotli: 2.21kb
│ .svelte/output/client/_app/pages/[list]/[page].svelte-f95225fd.js           6.05kb / brotli: 2.16kb
│ .svelte/output/client/_app/assets/pages/item/[id].svelte-28003fea.css       1.39kb / brotli: 0.37kb
│ .svelte/output/client/_app/assets/start-475c907e.css                        2.73kb / brotli: 0.89kb
│ .svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
│ .svelte/output/client/_app/chunks/vendor-b3a70cfd.js                        5.95kb / brotli: 2.28kb
│ .svelte/output/client/_app/start-8804a1d2.js                                23.38kb / brotli: 7.30kb
│ vite v2.1.5 building SSR bundle for production...
│ transforming...
│ ✓ 34 modules transformed.
│ rendering chunks...
│ .svelte/output/server/app.js   110.70kb
│ Run npm start to try your app locally.
│ > Using @sveltejs/adapter-static
│ > 500 /jobs/12
│ Error: 500 /jobs/12
│     at error (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.js:526:
│     at visit (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.js:581:
│     at processTicksAndRejections (internal/process/task_queues.js:93:5)
│     at async visit (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.j
│     at async visit (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.j
│     at async visit (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.j
│     at async visit (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.j
│     at async visit (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.j
│     at async visit (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.j
│     at async visit (file:///home/user/kit/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.71_396d186e69d9be4f41fb81a15b459f1b/node_modules/@sveltejs/kit/dist/chunks/index6.j
└─ Failed in 11.6s
../../packages/kit build$ rm -rf assets/runtime && rollup -c
[1 lines collapsed]
│ created assets/runtime in 181ms
│ src/cli.js, src/runtime/server/index.js, src/core/filesystem/index.js, src/core/http/index.js → dist...
│ (!) Unused external imports
│ default imported from external module 'url' but never used
│ default imported from external module 'querystring' but never used
│ (!) Circular dependency
│ src/runtime/server/index.js -> src/runtime/server/page/index.js -> src/runtime/server/index.js
│ created dist in 4.7s
│ src/core/filesystem/index.js → dist/filesystem.cjs...
│ created dist/filesystem.cjs in 10ms
└─ Done in 5.1s
 ERROR  hn.svelte.dev@1.0.0 build: `svelte-kit build`
Exit status 1
user@debian:~/kit/examples/hn.svelte.dev$ 
```

**Comment by pablodelfante** [comment]
The same happened to me

**Comment by benmccann** [comment]
https://github.com/sveltejs/kit/issues/982 would probably help a lot in figuring out the issue

**Comment by phaleth** [comment]
The `@sveltejs/adapter-static@1.0.0-next.5` release fixes the `build` step when deploying with docker....
```shell
user@debian:~/svelte-kit$ docker build --no-cache -t svelte-kit-node:2021-04-23 .
Sending build context to Docker daemon  81.92kB
Step 1/6 : FROM node:15-alpine
 ---> 75631da67663
Step 2/6 : WORKDIR /usr/src/app
 ---> Running in 3bb587034080
Removing intermediate container 3bb587034080
 ---> 205dc75cf4ff
Step 3/6 : COPY ./hn.svelte.dev .
 ---> d88ced072588
Step 4/6 : RUN npm i
 ---> Running in 15f79be79241

added 36 packages, and audited 37 packages in 6s

found 0 vulnerabilities
npm notice 
npm notice New minor version of npm available! 7.7.6 -> 7.11.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.11.0>
npm notice Run `npm install -g npm@7.11.0` to update!
npm notice 
Removing intermediate container 15f79be79241
 ---> 8ed678675f11
Step 5/6 : RUN npm run build
 ---> Running in e0c320c57f91

> hn.svelte.dev@1.0.0 build
> svelte-kit build --verbose

vite v2.2.1 building for production...
transforming...
✓ 35 modules transformed.
rendering chunks...
.svelte/output/client/_app/assets/fold.08edd659.svg                         0.44kb
.svelte/output/client/_app/assets/unfold.a370f815.svg                       0.43kb
.svelte/output/client/_app/manifest.json                                    2.54kb
.svelte/output/client/_app/pages/index.svelte-021be8bb.js                   0.21kb / brotli: 0.14kb
.svelte/output/client/_app/pages/about.svelte-91750a3f.js                   2.96kb / brotli: 1.04kb
.svelte/output/client/_app/pages/$error.svelte-34698e0e.js                  1.18kb / brotli: 0.58kb
.svelte/output/client/_app/pages/item/[id].svelte-6118d4c0.js               6.71kb / brotli: 2.22kb
.svelte/output/client/_app/pages/user/[name].svelte-9dfa866f.js             2.91kb / brotli: 1.19kb
.svelte/output/client/_app/pages/[list]/[page].svelte-a2c81547.js           6.04kb / brotli: 2.16kb
.svelte/output/client/_app/assets/start-a8cd1609.css                        0.16kb / brotli: 0.10kb
.svelte/output/client/_app/pages/$layout.svelte-b4f75cef.js                 7.57kb / brotli: 2.39kb
.svelte/output/client/_app/assets/pages/$error.svelte-3fddf370.css          0.20kb / brotli: 0.10kb
.svelte/output/client/_app/assets/pages/$layout.svelte-0143f758.css         2.39kb / brotli: 0.78kb
.svelte/output/client/_app/assets/pages/[list]/[page].svelte-c7a38061.css   0.56kb / brotli: 0.17kb
.svelte/output/client/_app/assets/pages/item/[id].svelte-db08ebba.css       1.41kb / brotli: 0.39kb
.svelte/output/client/_app/chunks/vendor-9feae743.js                        5.96kb / brotli: 2.29kb
.svelte/output/client/_app/start-e5b72958.js                                16.91kb / brotli: 5.34kb
vite v2.2.1 building SSR bundle for production...
transforming...
✓ 34 modules transformed.
rendering chunks...
.svelte/output/server/app.js   85.71kb

Run npm start to try your app locally.

> Using @sveltejs/adapter-static
  200 /about
  ✔ done
Removing intermediate container e0c320c57f91
 ---> 116eea9170cf
Step 6/6 : CMD ["node", "./build/index.js"]
 ---> Running in a58a93bfc4e5
Removing intermediate container a58a93bfc4e5
 ---> c503b4788da0
Successfully built c503b4788da0
Successfully tagged svelte-kit-node:2021-04-23
user@debian:~/svelte-kit$ 
```

...but now the container keeps restarting with the following error.
```shell
svelte-kit_1  | node:internal/modules/cjs/loader:927
svelte-kit_1  |   throw err;
svelte-kit_1  |   ^
svelte-kit_1  | 
svelte-kit_1  | Error: Cannot find module '/usr/src/app/build/index.js'
svelte-kit_1  |     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:924:15)
svelte-kit_1  |     at Function.Module._load (node:internal/modules/cjs/loader:769:27)
svelte-kit_1  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
svelte-kit_1  |     at node:internal/main/run_main_module:17:47 {
svelte-kit_1  |   code: 'MODULE_NOT_FOUND',
svelte-kit_1  |   requireStack: []
svelte-kit_1  | }
dc-01_svelte-kit_1 exited with code 1
svelte-kit_1  | node:internal/modules/cjs/loader:927
svelte-kit_1  |   throw err;
svelte-kit_1  |   ^
svelte-kit_1  | 
svelte-kit_1  | Error: Cannot find module '/usr/src/app/build/index.js'
svelte-kit_1  |     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:924:15)
svelte-kit_1  |     at Function.Module._load (node:internal/modules/cjs/loader:769:27)
svelte-kit_1  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
svelte-kit_1  |     at node:internal/main/run_main_module:17:47 {
svelte-kit_1  |   code: 'MODULE_NOT_FOUND',
svelte-kit_1  |   requireStack: []
svelte-kit_1  | }
dc-01_svelte-kit_1 exited with code 1
svelte-kit_1  | node:internal/modules/cjs/loader:927
svelte-kit_1  |   throw err;
svelte-kit_1  |   ^
svelte-kit_1  | 
svelte-kit_1  | Error: Cannot find module '/usr/src/app/build/index.js'
svelte-kit_1  |     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:924:15)
svelte-kit_1  |     at Function.Module._load (node:internal/modules/cjs/loader:769:27)
svelte-kit_1  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
svelte-kit_1  |     at node:internal/main/run_main_module:17:47 {
svelte-kit_1  |   code: 'MODULE_NOT_FOUND',
svelte-kit_1  |   requireStack: []
svelte-kit_1  | }
dc-01_svelte-kit_1 exited with code 1
```

I'm closing this issue and will investigate further.
